akka-http/server/src/main/scala/endpoints4s/akkahttp/server/BasicAuthentication.scala

```diff
@@ -77,14 +51,24 @@ trait BasicAuthentication extends algebra.BasicAuthentication with EndpointsWith
                 )
               )
             )
-          case validatedUrlAndHeaders => Directives.provide(validatedUrlAndHeaders)
+          case Valid(Some(credentials)) =>
+            matchAndProvideParsedUrlAndHeadersData(method, url, headers)
+              .map(_.map { case (u, h) => (u, h, credentials) })
```

Comment:
It is a little bit of a nitpick. But I find nested maps quite hard to read without explicitly named arguments. I would replace the underscore with a sensible name.

Reply:
That’s a good point! Addressed in add6b5c.